packages/payment-detection/src/eth/input-data.ts

```diff
@@ -156,23 +192,56 @@ export default class PaymentNetworkETHInputData
    * @param eventName Indicate if it is an address for payment or refund
    * @param network The id of network we want to check
    * @param paymentReference The reference to identify the payment
+   * @param paymentNetworkVersion the version of the payment network
    * @returns The balance
    */
   private async extractBalanceAndEvents(
     address: string,
     eventName: PaymentTypes.EVENTS_NAMES,
     network: string,
     paymentReference: string,
+    paymentNetworkVersion: string,
   ): Promise<PaymentTypes.ETHBalanceWithEvents> {
+    if (!PROXY_CONTRACT_ADDRESS_BY_VERSION_BY_NETWORK[paymentNetworkVersion]) {
+      throw new Error(`Payment network version not supported: ${paymentNetworkVersion}`);
+    }
+
+    const proxyContractAddress: string | undefined =
+      PROXY_CONTRACT_ADDRESS_BY_VERSION_BY_NETWORK[paymentNetworkVersion][network].address;
+    const proxyCreationBlockNumber: number =
+      PROXY_CONTRACT_ADDRESS_BY_VERSION_BY_NETWORK[paymentNetworkVersion][network]
+        .creationBlockNumber;
+
+    if (!proxyContractAddress) {
```

Comment:
the previous lines will throw before getting here, because if `network` is not supported `PROXY_CONTRACT_ADDRESS_BY_VERSION_BY_NETWORK[paymentNetworkVersion][network]` will be undefined, and trying to access one of it's properties will throw.

Would be better to check if `PROXY_CONTRACT_ADDRESS_BY_VERSION_BY_NETWORK[paymentNetworkVersion][network]` is defined before the assignments.